src/bin/pg_autoctl/supervisor.h

```diff
@@ -108,19 +119,49 @@ typedef struct Supervisor
 {
 	Service *services;
 	int serviceCount;
+
 	char pidfile[MAXPGPATH];
 	pid_t pid;
 	bool cleanExit;
 	bool shutdownSequenceInProgress;
 	int shutdownSignal;
 	int stoppingLoopCounter;
-} Supervisor;
 
+	/* dynamic handler function */
+	int (*dynamicHandler)(struct Supervisor *, void *);
+	void *dynamicHandlerArg;    /* dynamic handler private data */
 
-bool supervisor_start(Service services[], int serviceCount, const char *pidfile);
+	/*
+	 * Anonymous struct holding active dynamic services.
+	 * Do not access directly.
+	 */
+	struct
+	{
+		Service services[MAXDYNSERVICES];
+		int serviceCount;
+	} dynamicServices;
+
+	/*
+	 * Anonymous struct holding recently removed dynamic services.
+	 * Do not access directly.
+	 */
+	struct
+	{
+		Service services[MAXDYNSERVICES];
+		int serviceCount;
+	} dynamicRecentlyRemoved;
```

Comment:
Same comment as before, let's expose the currently internal and anonymous structure. And rename it, too.